man/hexxed-tutorial.7.scd

```diff
@@ -0,0 +1,156 @@
+*isprint*(3) for more information.
+
+# Step 2: Navigating
+
+Navigating in Hexxed is straightforward, use the arrow keys to move around the
+buffer and *Page up* and *Page down* to advance by the current terminal height.
+The *Home* and *End* keys go to the beginning and end of the buffer,
+respectively. Hexxed uses an overscroll of 3 rows.
+
+# Step 3: Selecting Blocks
+
+To select a block of *data*, use the *v* key. Hit *Home* or scroll to the start
+of the buffer, and hit *v* then move to the end of the line. Then, hit *v*
+again and move to the right. The cursor will be placed on the next line.
+
+```
+example                               UNK+.00000000`00000000
+00000000: [7f 45 4c 46-02 01 01 00-00 00 00 00] .ELF........
+0000000c: {02}00 3e 00-01 00 00 00-a0 12 40 00  ..>.......@.
+00000018:  40 00 00 00-00 00 00 00-80 ad 01 00  @...........
+00000024:  00 00 00 00-40 00 38 00-0d 00 40 00  ....@.8...@.
+00000030:  06 00 00 00-04 00 00 00-40 00 00 00  ........@...
+0000003c:  40 00 40 00-00 00 00 00-40 00 40 00  @.@.....@.@.
+00000048:  d8 02 00 00-00 00 00 00-d8 02 00 00  ............
+00000054:  08 00 00 00-00 00 00 00-03 00 00 00  ............
+00000060:  18 03 00 00-00 00 00 00-18 03 40 00  ..........@.
+0000006c:  18 03 40 00-00 00 00 00-1c 00 00 00  ..@.........
+00000078:  1c 00 00 00-00 00 00 00-01 00 00 00  ............
+00000084:  01 00 00 00-04 00 00 00-00 00 00 00  ............
+00000090:  00 00 40 00-00 00 00 00-00 00 40 00  ..@.......@.
+0000009c:  e0 0d 00 00-00 00 00 00-e0 0d 00 00  ............
+000000a8:  00 10 00 00-00 00 00 00-01 00 00 00  ............
+000000b4:  00 10 00 00-00 00 00 00-00 10 40 00  ..........@.
+000000c0:  00 10 40 00-00 00 00 00-45 0b 01 00  ..@.....E...
+000000cc:  45 0b 01 00-00 00 00 00-00 10 00 00  E...........
+000000d8:  01 00 00 00-04 00 00 00-00 20 01 00  ......... ..
+000000e4:  00 20 41 00-00 00 00 00-00 20 41 00  . A...... A.
+000000f0:  b8 3e 00 00-00 00 00 00-b8 3e 00 00  .>.......>..
+000000fc:  00 10 00 00-00 00 00 00-01 00 00 00  ............
+```
+
+- The *[]* enclosed characters are the selected *block*
+- The *{}* enclosed characters denote the cursor location
+
+# Step 4: Changing Representations
+
+In addition to the Hex representation, Hexxed also offers a *Raw* view. Since
+the panes share a buffer, they will also share underyling *comments* and blocks.
+Hit *Enter* to cycle panes into the *Raw* pane. Here, only printable characters
+are displayed on the screen. See *isprint(3)* for more information. Hit *Enter*
+to cycle back to the *Hex* view
+
+# Step 5: Editing
+
+Hexxed is fundamentally a hex editor, so it is straightforward to perform
+edit operations. Hit *F3* to enter edit mode, and the cursor will change to select
+a single *nibble* instead of a *pair*. To overwrite the data under the cursor,
+enter a character in the range *0-9* or *a-f*. The change will be reflected in the
+buffer. Hit *Escape* to exit.
```

Comment:
Do you think it should sabe clearer that changes will be written to the file? I don't know, I feel like hitting **Escape** suggests "operation cancelled" in a certain way but maybe it's just me. 🙂

Reply:
You make a good point, I should rewrite the edit mode to undo edits if Escape is hit, and save edits if F3 is hit again (and exit edit mode). Does this make more sense?

Reply:
Yes, I think so. I tried to look for a pattern, but apparently there is none. =/

[HIEW ](http://hiew.ru/)uses `F3` to enter editing mode:

![hiew_before](https://user-images.githubusercontent.com/1429647/104845670-61ea4500-58b5-11eb-99e3-c1a410c5bef1.png)

But then hitting `F3` again will undo the edits. `Escape` also undo them. Only `F9` saves the changes.

![hiew_edit_mode](https://user-images.githubusercontent.com/1429647/104845679-73cbe800-58b5-11eb-9d93-c53beb199e97.png)

[HT Editor](http://hte.sourceforge.net/) uses `F4` for entering editing mode:

![hte_before](https://user-images.githubusercontent.com/1429647/104845744-d45b2500-58b5-11eb-8cd9-ccbffb324067.png)

 `Escape` undo the changes while `F2` save them:

![hte_editing_mode](https://user-images.githubusercontent.com/1429647/104845776-f5237a80-58b5-11eb-94c6-4be7e74ae0cd.png)

In general, I'd suggest to try your best to make my shortcut keys compatible to existent software but in that case is a hard decision. HIEW is famous in malware analysis field, but it's paid. HT is open source, but it seems to be abandoned. hexxed is our new hope. :)

Reply:
Will start tracking this here: https://github.com/meme/hexxed/issues/8. Thanks for providing a comparison with other editors.